src/data/real/sqrt.lean

```diff
@@ -157,6 +157,21 @@ by rw [← abs_mul_abs_self x, sqrt_mul_self (abs_nonneg _)]
 theorem sqrt_sqr_eq_abs (x : ℝ) : sqrt (x ^ 2) = abs x :=
 by rw [pow_two, sqrt_mul_self_eq_abs]
 
+theorem sqr_abs : (abs x) ^ 2 = x ^ 2 :=
+by rw [← sqrt_sqr_eq_abs, sqr_sqrt (pow_two_nonneg x)]
+
+theorem sqr_lt_sqr (h : abs x < y) : x ^ 2 < y ^ 2 :=
+by simpa only [sqr_abs] using pow_lt_pow_of_lt_left h (abs_nonneg x) zero_lt_two
+
+theorem sqr_lt_sqr' (h1 : -y < x) (h2 : x < y) : x ^ 2 < y ^ 2 :=
+sqr_lt_sqr (abs_lt.mpr ⟨h1, h2⟩)
+
+theorem sqr_le_sqr (h : abs x ≤  y) : x ^ 2 ≤ y ^ 2 :=
+by simpa only [sqr_abs] using pow_le_pow_of_le_left (abs_nonneg x) h 2
+
+theorem sqr_le_sqr' (h1 : -y ≤ x) (h2 : x ≤ y) : x ^ 2 ≤ y ^ 2 :=
+sqr_le_sqr (abs_le.mpr ⟨h1, h2⟩)
+
```

Comment:
These lemmas don't mention the square root at all in their statements; their proofs can probably be rejiggered to avoid mention of the square root, and they can move to `algebra.ordered_group` or `algebra.ordered_ring` or similar.  Indeed, I'm pretty sure I've seen some of these lemmas there already ... I'll take a look.

Reply:
Are you suggesting that they should be moved to that file and deduced from the lemmas there, or that they are redundant because of their similarity to lemmas contained in that file?

Reply:
The former!